sdk/core/core-lro/docs/LROEngine.md

```diff
@@ -0,0 +1,214 @@
+# Modular Support for Long-Running Operations
+
+Long-running operations (LROs) are operations that the service _could_ take a long time to finish processing and they follow a common convention:
+
+- the customer first send a an initiation request to the service, which in turn sends back a response, from which the customer can learn how to poll for the status of the operation, if it has not been completed already,
+- using their learnings, the customer polls the status of the operation until it is done,
+- again, using their learnings, the customer can now get the desired result of the operation once its status says it is done.
+
+Ideally, we can write an algorithm that implements this convention once and use it in all Azure clients for LRO APIs, however, in reality, this convention is implemented differently across Azure services. The good news is that the TypeScript Autorest extension is AFAIK able to generate code that implements those different ones, but this implementation has a couple limitations:
+
+1. it is located in a few files that the code generator copies into every generated package that has LROs. So if in the future there is a bug fix needed in the LRO logic, the package has to be regenerated with the fix.
+2. it works only with clients that use `@azure/core-client`, so clients that use `@azure-rest/core-client` or `@azure/core-http` can not use this implementation as is.
+
+In order to fix limitation #1, the most straightforward thing to do is to move those files into `@azure/core-lro`, but without fixing limitation #2 first, `@azure/core-lro` will have to depend on `@azure/core-client` in this case which will force clients that depend on `@azure/core-lro` but not necessarily depend on `@azure/core-client` to transitively depend on the latter, posing concerns about unnecessarily larger bundle sizes.
+
+This document presents a design that fixes limitation #2 and naturally fixes limitation #1 too.
```

Comment:
While I appreciate the context, I’m not sure if this design proposal should be in the repository at the end. I would expect the design proposal to be somewhere else, and the docs in the core-lro repo to be about documentation.

Reply:
We did that before for core v2: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-rest-pipeline/documentation/core2.md. However, I am open to move/remove it if there is a consensus on this.

Reply:
oh ok ok! I’m ok leaving this here then. 👍